SME.CDEP.Aplicacao/Servicos/ServicoConservacao.cs

```diff
@@ -0,0 +1,50 @@
+﻿using AutoMapper;
+using SME.CDEP.Aplicacao.DTOS;
+using SME.CDEP.Aplicacao.Servicos.Interface;
+using SME.CDEP.Dominio.Entidades;
+using SME.CDEP.Infra.Dados.Repositorios.Interfaces;
+
+namespace SME.CDEP.Aplicacao.Servicos
+{
+    public class ServicoConservacao : IServicoConservacao
+    {
+        private readonly IRepositorioConservacao repositorioConservacao;
+        private readonly IMapper mapper;
+        
+        public ServicoConservacao(IRepositorioConservacao repositorioConservacao, IMapper mapper) 
+        {
+            this.repositorioConservacao = repositorioConservacao ?? throw new ArgumentNullException(nameof(repositorioConservacao));
+            this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        public async Task<long> Inserir(ConservacaoDTO conservacaoDTO)
+        {
+            var conservacao = mapper.Map<Conservacao>(conservacaoDTO);
+            return await repositorioConservacao.Inserir(conservacao);
+        }
+
+        public async Task<IList<ConservacaoDTO>> ObterTodos()
+        {
+            return (await repositorioConservacao.ObterTodos()).ToList().Where(w=> !w.Excluido).Select(s=> mapper.Map<ConservacaoDTO>(s)).ToList();
```

Comment:
Remover async e await
o primeiro .ToList() não faz sentido pois o where e select geram um IEnumerable novamente.
```suggestion
        public async Task<IList<ConservacaoDTO>> ObterTodos()
        {
            return (await repositorioConservacao.ObterTodos()).Where(w=> !w.Excluido).Select(s=> mapper.Map<ConservacaoDTO>(s)).ToList();
```

Reply:
Ajustado